src/metabase/driver/postgres.clj

```diff
@@ -982,19 +982,48 @@
   [driver]
   (= driver :postgres))
 
-(defmethod sql-jdbc.sync/alter-columns-sql :postgres
-  [driver table-name column-definitions]
+(defn- alter-column-using-hsql-expr
+  "Returns a honey expr to convert the column from the old type to the new when a USING expr is necessary (nil if not).
+
+  The output can be used in the context of ALTER COLUMN when the :old-type key is provided
+  as part of the new column-definition.
+
+  Unfortunately out of the box not all trivial conversions are possible using CAST alone:
+  e.g. BOOLEAN to BIGINT, or BOOLEAN to FLOAT.
+  This function therefore provides more options for alter-columns-sql to produce DDL that can 'just work' without
+  extra input from the user."
+  [column new-type old-type]
+  (condp = [new-type old-type]
+    [[:bigint] [:boolean]]
+    [:case
+     (quote-identifier column) 1
+     :else 0]
+
+    [[:float] [:boolean]]
+    [:case
+     (quote-identifier column) 1.0
+     :else 0.0]
+
+    nil))
```

Comment:
Further to my other comment, I'm a bit nervous about these "open" semantics, where we assume all other pairs can be implicitly converted. What do you think about just enumerating them with a sentinel value like `true`?

Reply:
Checkout the new docstring [f0e7d05](https://github.com/metabase/metabase/pull/52607/commits/f0e7d05ebbfa0bac55cd8013d630c29654e71959)

Not sure if this is satisfactory or not for now, trying to be more transparent about the limitations and implicit nature here. Feel like this fn will change anyway

Reply:
> if one takes the implementation to be transparent in terms of alter column limitations what we have here is fine

Where my thinking was going is that we could make this method the source of truth for what is supported